targets/drive/mobile/main.jsx

```diff
@@ -38,78 +43,80 @@ if (__DEVELOPMENT__) {
 }
 
 // Register callback for when the app is launched through cozydrive:// link
-window.handleOpenURL = require('drive/mobile/lib/handleDeepLink').default(hashHistory)
-
-const renderAppWithPersistedState = (persistedState = {}) => {
-  const cozyURL = persistedState.mobile
-    ? persistedState.mobile.settings.serverUrl
-    : ''
-  configureReporter()
-  const client = initClient(cozyURL)
-  const store = configureStore(client, persistedState)
-
-  const clientInfos = store.getState().settings.client
-  if (clientInfos) {
-    client.isRegistered(clientInfos).then(isRegistered => {
-      if (isRegistered) {
-        startReplication(store.dispatch, store.getState) // don't like to pass `store.dispatch` and `store.getState` as parameters, big coupling
-        initBar(client)
-      } else {
-        console.warn('Your device is no more connected to your server')
-        store.dispatch(revokeClient())
-      }
+window.handleOpenURL = require('drive/mobile/lib/handleDeepLink').default(
+  hashHistory
+)
+
+function uploadIntent(intent, store) {
```

Comment:
Here is the code that might be refactored, it is ugly and to verbose.
It includes so many responsibilities that we don't understand what are the steps involved!
A new commit (only refactoring) is coming.